### PR TITLE
feat(kumactl): add MeshGateway to `install demo`

### PR DESCRIPTION
**Pull request by michaelbeaumont**
https://github.com/kumahq/kuma-counter-demo/issues/28

### Checklist prior to review

Each of these sections need to be filled by the author when opening the PR.

If something doesn't apply please check the box and add a justification after the `--`

- [x] Link to docs PR or issue -- https://github.com/kumahq/kuma-website/issues/916
- [x] Link to UI issue or PR -- 
- [x] Is the [issue worked on linked][1]? --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kumactl/cmd/install/testdata/install-demo.defaults.golden.yaml [hunk 1]
```diff
@@ -91,3 +91,49 @@ spec:
...
+      kuma.io/service: demo-app_gateway
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshGatewayInstance
+metadata:
+  name: demo-app-gateway
+  namespace: kuma-demo
+spec:
+  replicas: 1
+  serviceType: LoadBalancer
```
Comment (lahabana) [inline app/kumactl/cmd/install/testdata/install-demo.defaults.golden.yaml:116]: This works without problems in k3d even if there's no LB types?

**Review by lahabana — CHANGES_REQUESTED** [review]
The file app/kumactl/data/instal/k8s/demo/gateway.yaml is meant to be updated with:

https://github.com/kumahq/kuma/blob/778b94ab1a7e24bf254e28557727de825ba94b3f/mk/dev.mk#L348-L353

So you need to add this to kumahq/kuma-counter-demo .

File: app/kumactl/cmd/install/testdata/install-demo.defaults.golden.yaml (hunk 1 above)
Comment (michaelbeaumont) [reply app/kumactl/cmd/install/testdata/install-demo.defaults.golden.yaml:116]: If there's no LB support then it just behaves like a ClusterIP service.

Comment (michaelbeaumont) [reply app/kumactl/cmd/install/testdata/install-demo.defaults.golden.yaml:116]: https://github.com/kumahq/kuma-counter-demo/pull/29

**Review by lahabana — APPROVED** [review]

**Comment by lahabana** [comment]
Probably want to link to a doc update no?

**Comment by michaelbeaumont** [comment]
@lahabana to update the generated docs?

**Comment by lahabana** [comment]
No just update the getting started docs on k8s to talk about gateway

**Pull request merged**